gptel-openai.el

```diff
@@ -330,28 +341,49 @@ Mutate state INFO with response metadata."
+                   (let* ((tool-call (read (current-buffer)))
+                          (id (gptel--openai-format-tool-id id))
+                          (name (plist-get tool-call :name))
+                          (arguments (json-serialize (plist-get tool-call :args)
+                                                     :null-object nil
+                                                     :false-object :json-false)))
+                     (push (list :role "tool"
+                                 :tool_call_id id
+                                 :content
+                                 (string-trim
+                                  (buffer-substring-no-properties (point) prev-pt)))
+                           prompts)
+                     (push (list :role "assistant"
+                                 :tool_calls
+                                 (vector (list :type "function" :id id
+                                               :function `( :name ,name
+                                                            :arguments ,arguments))))
+                           prompts))
+                 ((end-of-file invalid-read-syntax)
+                  (message (format "Could not parse tool-call %s on line %s"
+                                   id (line-number-at-pos (point))))))))
+            ('ignore)
             ('nil
+             (and max-entries (cl-decf max-entries))
              (if include-media
-                 (push (list :role "user"
-                             :content
-                             (gptel--openai-parse-multipart
-                              (gptel--parse-media-links major-mode (point) prev-pt)))
-                       prompts)
-               (push (list :role "user"
-                           :content
-                           (gptel--trim-prefixes
-                            (buffer-substring-no-properties (point) prev-pt)))
-                     prompts))))
-          (setq prev-pt (point))
-          (and max-entries (cl-decf max-entries)))
-      (push (list :role "user"
-                  :content
-                  (gptel--trim-prefixes (buffer-substring-no-properties
-                                         (point-min) (point-max))))
-            prompts))
+                 (when-let* ((content (gptel--openai-parse-multipart
+                                       (gptel--parse-media-links major-mode
+                                                                 (point) prev-pt))))
+                   (when (length content)
+                     (push (list :role "user" :content content) prompts)))
+               (when-let* ((content (gptel--trim-prefixes (buffer-substring-no-properties
+                                                           (point) prev-pt))))
+                 (push (list :role "user" :content content) prompts)))))
+          (setq prev-pt (point)))
+      (let* ((content (string-trim (buffer-substring-no-properties
```

Comment:
This `let*` does nothing, please revert this part.

Reply:
Generally, non-nested expressions are easier to read.  I did remove this redundant star but I retained both let expressions.